libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -123,8 +123,7 @@ func New(cfg *config.C) (beat.Processor, error) {
 }
 
 func newProcessorConfig(cfg *config.C, register *Register) (kubeAnnotatorConfig, error) {
-	config := defaultKubernetesAnnotatorConfig()
-
+	var config kubeAnnotatorConfig
```

Comment:
Upon construction of the config object, the InitDefaults is called under the hood?

Reply:
yes, https://github.com/elastic/go-ucfg/blob/main/initializer.go#L24-L29

[`Unpack`](https://github.com/elastic/go-ucfg/blob/main/reify.go#L137C18-L137C24) calls [reifyInfo](https://github.com/elastic/go-ucfg/blob/main/reify.go#L155), where after detecting the type (Struct) will be called function [`tryInitDefaults`](https://github.com/elastic/go-ucfg/blob/main/reify.go#L261)

It is not a required changed, but I've noticed that it is used in elastic-agent and in beats in multiple places

Reply:
yes, https://github.com/elastic/go-ucfg/blob/main/initializer.go#L24-L29

[`Unpack`](https://github.com/elastic/go-ucfg/blob/main/reify.go#L137C18-L137C24) calls [reifyInfo](https://github.com/elastic/go-ucfg/blob/main/reify.go#L155), where after detecting the type (Struct) will be called function [`tryInitDefaults`](https://github.com/elastic/go-ucfg/blob/main/reify.go#L261)

It is not a required changed, but I've noticed that it is used in elastic-agent and in beats in multiple places